packages/@aws-cdk/aws-rds/README.md

```diff
@@ -13,87 +14,125 @@
 ---
 <!--END STABILITY BANNER-->
 
-### Starting a Clustered Database
+```typescript
+import * as rds from '@aws-cdk/aws-rds';
+```
+
+### Starting a clustered database
 
 To set up a clustered database (like Aurora), define a `DatabaseCluster`. You must
 always launch a database in a VPC. Use the `vpcSubnets` attribute to control whether
 your instances will be launched privately or publicly:
 
 ```ts
-const cluster = new DatabaseCluster(this, 'Database', {
-    engine: DatabaseClusterEngine.AURORA,
-    masterUser: {
-        username: 'clusteradmin'
+const cluster = new rds.DatabaseCluster(this, 'Database', {
+  engine: rds.DatabaseClusterEngine.AURORA,
+  masterUser: {
+    username: 'clusteradmin'
+  },
+  instanceProps: {
+    // optional, defaults to the engine's default instance type
+    instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
+    vpcSubnets: {
+        subnetType: ec2.SubnetType.PUBLIC,
     },
-    instanceProps: {
-        instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
-        vpcSubnets: {
-            subnetType: ec2.SubnetType.PUBLIC,
-        },
-        vpc
-    }
+    vpc
+  }
 });
 ```
+
+To use a specific version of the engine
+(which is recommended, in order to avoid surprise updates when RDS add support for a newer version of the engine),
+use instance methods on the passed `engine`:
+
+```typescript
+new rds.DatabaseCluster(this, 'Database', {
+  engine: rds.DatabaseClusterEngine.AURORA.withVersion1dot17dot9(),
```

Comment:
Aaaargh my eyes! 

```suggestion
  // Get Romain to change jsii to allow this
  engine: rds.DatabaseClusterEngine.AURORA.withVersion_1_17_9(),
```

```suggestion
  engine: rds.DatabaseClusterEngine.AURORA.VERSION_1_17_9,
```

```suggestion
  engine: rds.DatabaseClusterEngine.AURORA.withVersion('1.17.9'),
```

```suggestion
  engine: rds.DatabaseClusterEngine.AURORA.withVersion(rds.AuroraVersion.VERSION_1_17_9),
```

Reply:
Better yet:

```ts
engine: rds.DatabaseClusterEngine.AURORA_LATEST,
// or
engine: rds.DatabaseClusterEngine.AURORA_1_17_9,
// or
engine: rds.DatabaseClusterEngine.auroraVersion('6.7.8'),
```

Reply:
For the record, my preference is:
```ts
rds.DatabaseClusterEngine.AURORA_1_17_9,
```

---

> However, I will not have any `LATEST` here.

Yep, `LATEST` is problems waiting to happen. The meaning of `LATEST` is likely going to be "as of the currently used construct library version", which is likely not what a user would want to mean.

---

Allowing underscores in method names is a slippery slope and will likely cause a myriad of problems, not to mention questionnable ergonomics in some/all other languages. Definitely not in favor of that option.

Reply:
There is a major problem with this solution though.

At some moment, we will have to introduce [engine modes](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html#cfn-rds-dbcluster-enginemode). What happens then? Do we have an `AURORA_1_17_9_SERVERLESS`, `AURORA_1_17_9_PARALLEL_QUERY`, and so on for every version-engine mode combination?

I wanted to avoid this explosions with methods like `withVersion()`, and probably adding `withEngingeMode()` later.

Thoughts on this? @rix0rrr @nija-at

Reply:
```shell
$ aws rds describe-db-engine-versions --query 'length(DBEngineVersions)'
425 # !!!
```
I think that in practice only `withVersion()` will be used. Also, for Oracle it can look like this `12.1.0.2.v20` but also like this `12.2.0.1.ru-2018-10.rur-2018-10.r1` or `19.0.0.0.ru-2020-04.rur-2020-04.r1` and I expect production database to be as specific as possible in the version.